arraycontext/impl/pyopencl/__init__.py

```diff
@@ -303,8 +296,7 @@ def call_loopy(self, t_unit, **kwargs):
 
     def clone(self):
         return type(self)(self.queue, self.allocator,
-                wait_event_queue_length=self._wait_event_queue_length,
-                force_device_scalars=self._force_device_scalars)
+                wait_event_queue_length=self._wait_event_queue_length)
```

Comment:
I think this needs a `force_device_scalars=None` to stay, so that subclasses that still have that argument get the expected result on clone? (hopefully also means no immediate changes in grudge / meshmode)

Reply:
Might also need to keep `self._force_device_scalars = None` in `__init__` (or as a deprecated property).

Reply:
Thanks! Yeah, looks like `self._force_device_scalars` needs to stick around for now, since subclasses can/do use it. I attempted to address the remaining `clone()` compatibility in https://github.com/inducer/arraycontext/pull/278/commits/88d33413c8b3885aaeb9f6cb0e972d53667a1c29.

Reply:
That looks good to me! Thanks!

It seems like all the downstream tests are passing now too :grin: